DomoAggregatorPlugin/DataReader.cs

```
@@ -150,96 +160,129 @@ public IList<string> GetHeaders()
+                // send the row data back in the same order as the headers
+                foreach (var header in GetHeaders())
                 {
-                    _readerProperties.QueryVariables[key] = _currentConnection.Reader[header].ToString();
-                    _callbackHost.SetReaderProperties(PropertyHelper.Serialize(_readerProperties));
+                    //Add data for additional data source column allowing us to determine where the query results originated from
+                    if (DatabaseSourceColumnName.Equals(header))
+                    {
+                        rowData.Add(_currentConnection.DSN);
+                        continue;
+                    }
+
+                    rowData.Add(_currentConnection.Reader[header]);
+
+                    var key = $"{_currentConnection.DSN}:{header}";
+                    if (_readerProperties.QueryVariables.ContainsKey(key) && _currentConnection.Reader[header] != null)
+                    {
+                        _readerProperties.QueryVariables[key] = _currentConnection.Reader[header].ToString();
+                        _callbackHost.SetReaderProperties(PropertyHelper.Serialize(_readerProperties));
+                    }
+
+                    if (_cancelRequested)
+                    {
+                        throw new OperationCanceledException();
+                    }
                 }
 
-                if (_cancelRequested)
-                {
-                    throw new OperationCanceledException();
-                }
-            }
+                //LogEvent(LogMessageType.Progress, "GetRowData End");
 
-            LogEvent(LogMessageType.Progress, "GetRowData End");
-
-            return rowData;
+                return rowData;
+            }
+            catch (Exception e)
+            {
+                EmailNotification.EmailNotificationSender(e.ToString(), "GetRowData");
+                return null;
```

Instead of returning null, we can re-throw the exception: `throw e;`

---

Good idea

---

Fixed

---

Fixed